src/shadowbox/server/manager_service.spec.ts

```diff
@@ -384,29 +385,28 @@ describe('ShadowsocksManagerService', () => {
 
   describe('setAccessKeyDataLimit', () => {
     it('sets access key limit', async (done) => {
+      const serverConfig = new InMemoryConfig({} as ServerConfigJson);
       const repo = getAccessKeyRepository();
-      const service = new ShadowsocksManagerService('default name', null, repo, null, null);
-      const accessKey = await repo.createNewAccessKey();
-      expect(accessKey.dataLimit).toBeUndefined();
-      expect(accessKey.isOverDataLimit()).toBeFalsy();
+      spyOn(repo, 'setAccessKeyDataLimit');
```

Comment:
I don't really see the point of this spy, we should test the external api that the new limit was set as opposed to testing the implementation

Reply:
Done - testing that the property is returned in `getServer`. The point of the spy is to ensure that the value is being passed to the repo; otherwise enforcement is not possible.